src/h3lib/include/vertex.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file  vertex.h
+ *  @brief Functions for working with cell vertexes.
+ */
+
+#ifndef H3VERTEX_H
+#define H3VERTEX_H
+
+#include "h3Index.h"
+#include "faceijk.h"
+
+/** @struct BaseCellRotation
+ *  @brief  The CCW rotations for some base cell on a given face
+ */
+typedef struct {
+    int face;      ///< face number
+    int ccwRot60;  ///< number of ccw 60 degree rotations for vertices
+                   ///  relative to base cell home face
+} BaseCellRotation;
```

Comment:
This looks very similar to the new `BaseCellOrient` struct. Is there a reason the logic couldn't be collapsed to one of the two? (Also, since this uses a face number instead of a base cell, this should probably be renamed to "FaceRotation" or something like that?)

Reply:
It's the reverse of `BaseCellOrient` - `BaseCellOrient` has a baseCell and a rotation, and it's indexed by face; this has a face and rotation, and it's indexed by base cell. The only ways to combine them would be:
- make one struct with both `face` and `baseCell`, which while it's fine logically would bloat the lookup tables by repeating information from the index
- make the struct have one property that could be either `face` or `baseCell`. I don't know what you'd call the property, and it seems likely to be more confusing.

Is there any particular benefit to combining them?

I agree though that the naming between the two is inconsistent - since this is really the complement to `BaseCellOrient`, I guess `FaceOrient` would make sense, though I like `Rotation` better. Maybe rename the first to `BaseCellRotation` and this one to `FaceRotation`? That seems a little weird, though, since it sounds like the face is rotated, so maybe `FaceOrient` would be best - open to suggestions.

Reply:
So based on what you're saying, the current `BaseCellOrient` would be a set of unique values (each base cell has only one rotation value associated with it) while `BaseCellRotation` wouldn't have that 1:1 mapping, since it's the set of rotations that can exist on that particular triangle face. And basically what you have here is similar to a set of DB indexes on three columns: baseCell, face, and rotation, and you're querying by face and baseCell, respectively, and getting the other two values out of it.

I don't know if this is an improvement or not, but how about:

`BaseCellOrient` => `BaseCellRotationPair`
`BaseCellRotation` => `FaceRotationPair`

Just simply calling them out as a pairing but not saying *why*. The data structure that *contains them* would have a more descriptive name, like `FaceRotationByBaseCell` and `BaseCellRotationsByFace` (the latter plural since you'd get an array of structs back.)

Reply:
That's not quite correct - `BaseCellOrient` is only unique _per face_ (i.e. for a face X and baseCell Y, there's one rotation). Same with `BaseCellRotation`, though due to some complexity around pentagons the unique face/baseCell pairs do not have the same rotation in these two maps (though I think they could, if I applied the extra pentagon rotations at runtime instead of in the table). 

I _think_ now that I fully understand the rotations required, I could just use the existing `faceIjkBaseCells` map at runtime, with a `O(27)` lookup, as long as I had a separate table to determine which face corresponded to which pentagon axial direction. But there would be some cost in speed (maybe negligible?) and code complexity. Does this seem worthwhile? In JS, I'd probably do it to avoid the extra weight of the lookup table - maybe it's worthwhile here to avoid having duplicative info in the code? I'd be interested in opinions here, not sure how to weigh the tradeoffs.

The `-Pair` naming seems potentially confusing to me, since one member of the "pair" is the index in the map, and the object itself doesn't look like a pair.

Reply:
Your answer here has confused me even more.

Why would they not be the pair, with the third value not in the struct being the index currently used?

I understand your point about multiple faces for base cells that sit on the edges, particularly pentagons, though.

Reply:
I was thinking the "pair" here is `baseCell/face` -`ccwRot60` is the value associated with the pair. In a language that supported a dictionary with tuple keys, you'd have that value indexed by `(baseCell, face)`. So one member of the pair is the index in both cases.

Reply:
OK, after all this, I've validated that I can use the `faceIjkBaseCells` table directly with no appreciable performance consequence - all I really need is a table for the face-to-direction lookup for each pentagon. So I'm actually thinking I ought to drop the new lookup table, switch my script to `generatePentagonFaceDirections`, and then I'd just be using `BaseCellOrient` directly. Then I don't need a new struct at all :).